File: tools_box/_hr/report/expense_claim_report/expense_claim_report.py
# Copyright (c) 2013, masonarmani38@gmail.com and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe


def execute(filters=None):
    if not filters: filters = {}
    columns = ["Date:Date:100", "Doc ID:Link/Expense Claim:150", "Expense Claim Type:Link/Expense Claim Type:100",
               "Description:Text:200", "Amount:Currency:120", "Cost Center:Link/Cost Center:130", "Approver:Link/User:200"]
    conditions, include_auth, comment_by = "", "", ""
    if filters.get("from_date"):
        conditions += " AND d.expense_date >= %(from_date)s"
    if filters.get("to_date"):
        conditions += " AND d.expense_date <= %(to_date)s"
    if filters.get("approver"):
        conditions += " AND p.exp_approver = %(approver)s"
    if filters.get("expense_type"):
        conditions += " AND d.expense_type = %(expense_type)s"
    if filters.get("cost_center"):
        conditions += " AND p.cost_center = %(cost_center)s"
    if filters.get("include_authorized"):
        comment_by = ",  c.comment_by"
        columns += [ "Authorized:Link/User:200"]
        include_auth = "LEFT OUTER JOIN tabComment c ON (p.name = c.comment_docname) and c.comment = 'Authorized'"

    data = frappe.db.sql(
        "SELECT d.expense_date,p.name,d.expense_type,d.description,d.sanctioned_amount,p.cost_center,p.exp_approver {0} FROM "
        "`tabExpense Claim` p LEFT JOIN `tabExpense Claim Detail` d ON (p.name = d.parent) {1} WHERE p.docstatus=1 {2}"
            .format(comment_by, include_auth, conditions), filters)

    return columns, data

File: tools_box/_hr/report/expense_claim_report/expense_claim_report.js
// Copyright (c) 2016, bobzz.zone@gmail.com, masonrmani38@gmail.colm and contributors
// For license information, please see license.txt

frappe.query_reports["Expense Claim Report"] = {
    "filters": [
        {
            fieldname: "from_date",
            label: __("From Date"),
            fieldtype: "Date",
        },
        {
            fieldname: "to_date",
            label: __("To Date"),
            fieldtype: "Date",
        },
        {
            fieldname: "approver",
            label: __("Approver"),
            fieldtype: "Link",
            options: "User"
        },
        {
            fieldname: "expense_type",
            label: __("Expense Claim Type"),
            fieldtype: "Link",
            options: "Expense Claim Type",
        },
        {
            fieldname: "cost_center",
            label: __("Cost Center"),
            fieldtype: "Link",
            options: "Cost Center",
        },
        {
            fieldname: "include_authorized",
            label: __("Include Authorizer"),
            fieldtype: "Check",
        },
    ],
    onload: function(report) {
        //query: "erpnext.hr.doctype.expense_claim.expense_claim.get_expense_approver"
    }
};
